packages/std/src/math/decimal.rs

```diff
@@ -205,6 +210,25 @@ impl ops::Sub for Decimal {
     }
 }
 
+impl ops::Mul for Decimal {
+    type Output = Self;
+
+    #[allow(clippy::suspicious_arithmetic_impl)]
+    fn mul(self, other: Self) -> Self {
+        // Decimals are fractions. We can multiplby two decimals a and b
+        // via
+        //       (a.numerator() * b.numerator()) / (a.denominator() * b.denominator())
+        //     = (a.numerator() * b.numerator()) / a.denominator() / b.denominator()
+
+        let result_as_uint256 =
+            self.numerator().full_mul(other.numerator()) / Self::DECIMAL_FRACTIONAL_UINT256;
+        let result: Uint128 = result_as_uint256
+            .try_into()
+            .expect("attempt to multiply with overflow");
+        Decimal(result)
+    }
+}
+
 /// Both d*u and u*d with d: Decimal and u: Uint128 returns an Uint128. There is no
 /// specific reason for this decision other than the initial use cases we have. If you
 /// need a Decimal result for the same calculation, use Decimal(d*u) or Decimal(u*d).
```

Comment:
Sorry for comment something not quite related to this PR, but it is `ops::Mul`.

From the comment in line 234 said, if I want d*u in Decimal, I should use `Decimal(d*u)`, but `Decimal(d*u)` doesn't produce a correct result.

For example, if I multiply 1.5 by 3, expected result should be 4.5. But `Decimal(d*u)` will return 4. Shall we have another method to cover this?

Reply:
That's a good question. We implemented Decimal * Uint128 = Uint128. So only integers are possible as a result (see the comment).

If we want Decimal * Uint128 = Decimal then this is another issue (currently not supported). Can you open a ticket for that?